src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -1065,6 +1094,49 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
         }
     }
 
+    void visit_Attribute(const AST::Attribute_t &x) {
+        if (AST::is_a<AST::Name_t>(*x.m_value)) {
+            std::string value = AST::down_cast<AST::Name_t>(x.m_value)->m_id;
+            ASR::symbol_t *t = current_scope->scope[value];
+            if (!t) {
+                throw SemanticError("'" + value + "' is not defined in the scope",
+                    x.base.base.loc);
+            }
+            if (ASR::is_a<ASR::Variable_t>(*t)) {
+                ASR::Variable_t *var = ASR::down_cast<ASR::Variable_t>(t);
+                LFORTRAN_ASSERT(var->m_value);
```

Comment:
Presently this fails on the following check:
```py
def test():
    x: c64
    x = 2 + 3j
    a: f64
    a = x.imag
```

ASR:
```
AssertFailed: var->m_value
```
This might be because we are keeping value as `nullptr` while creating a variable which I expect to be changed after the assignment. @certik What are your views/suggestions on this?